src/vs/workbench/contrib/workspace/browser/workspace.contribution.ts

```diff
@@ -165,25 +163,38 @@ export class WorkspaceTrustRequestHandler extends Disposable implements IWorkben
 			}
 
 			// Dialog
-			const result = await this.dialogService.show(
-				Severity.Info,
-				title,
-				buttons.map(b => b.label),
-				{
-					cancelId: buttons.findIndex(b => b.type === 'Cancel'),
-					custom: {
-						icon: Codicon.shield,
-						markdownDetails: [
-							{ markdown: new MarkdownString(message) },
-							{ markdown: new MarkdownString(localize('immediateTrustRequestLearnMore', "If you don't trust the authors of these files, we do not recommend continuing as the files may be malicious. See [our docs](https://aka.ms/vscode-workspace-trust) to learn more.")) }
-						]
+			const { result } = await this.dialogService.prompt({
```

Comment:
@lszomoru fyi please check, this one is a bit more complex adoption of the new dialog api